packages/frontend/web/src/pages/Register.tsx

```diff
@@ -0,0 +1,251 @@
+            />
+            <button
+              className='button border-dark m-1 w-[305px] rounded border-[5px] bg-blue-500 p-1 text-sm font-bold text-white hover:bg-blue-700 md:w-96 md:text-xl'
+              type='submit'
+            >
+              {'Register'}
+            </button>
+          </form>
+        </div>
+      </div>
+      <div className='fixed top-1 z-40 flex h-1/5 flex-col items-start justify-center'>
+        {email !== '' && !/^[\w-.]+@([\w-]+\.)+[\w-]{2,4}$/.test(email) && (
+          <div className='xxs:text-xs mb-3 mt-[70px] rounded border-2 border-red-600 bg-red-300 p-1 sm:text-base'>
+            {'example@example.com'}
+          </div>
+        )}
+        {password !== '' && password.length <= 10 && (
+          <div className='xxs:text-xs my-1 mb-3 rounded border-2 border-red-600 bg-red-300 p-1 sm:text-base'>
+            {'Password must be at least 10 characters long'}
+          </div>
+        )}
+        {confirmPassword !== '' && confirmPassword !== password && (
+          <div className='xxs:text-xs my-1 mb-3 rounded border-2 border-red-600 bg-red-300 p-1 sm:text-base'>
+            {'Confirm Password must match Password'}
+          </div>
+        )}
+        {pseudo !== '' && pseudo.length < 5 && (
+          <div className='xxs:text-xs my-1 mb-3 rounded border-2 border-red-600 bg-red-300 p-1 sm:text-base'>
+            {'Pseudo must be at least 5 characters long'}
+          </div>
+        )}
+      </div>
+      <div className='fixed top-1 z-40 flex h-1/5 flex-col items-center justify-center'>
+        {isSubmitted && birthdate !== '' && birthdate !== null ? (
+          birthdate && isUnderAge ? (
+            <div className='w-7/8 h-7/8 fixed top-12 flex rounded border-2 border-red-600 bg-red-300 p-1'>
+              {"Remember! No Booze 'til you're 18!"}
+            </div>
+          ) : (
+            <div className='bg-light-orange fixed top-0 z-50 flex h-full w-full items-start justify-center overflow-y-auto p-5'>
+              <div
+                className='z-50 flex h-screen w-screen items-center justify-center overflow-y-auto bg-contain bg-no-repeat'
+                style={{
+                  backgroundImage: `url('/yes-you-can.svg')`,
+                  backgroundPosition: 'center',
+                }}
+              >
+                <div
+                  className='animation-scale-up-delayed animate-fade-out z-40 mt-[32rem] h-screen w-screen overflow-x-hidden bg-center bg-no-repeat'
+                  style={{ backgroundImage: `url('/wow.svg')` }}
+                />
+              </div>
+            </div>
+          )
+        ) : null}
+      </div>
+      <div className='z-50'>
+        {isImageShown && isSubmitted && birthdate !== '' ? (
+          <div
+            className={`w-7/8 animate-fade-out ml-[5rem] flex h-screen items-center justify-center`}
+          >
+            <img
+              src='/ouch.svg'
+              alt='ouch! Not Eighteen.'
```

Comment:
Y'a pas la même chose dans `CheckBirthdate` ? Code dupliqué ?

Reply:
Toutafé. C'est ce que je m'attendais à ce que tu dises. Donc en gros, le rendu des animations de CheckBirthdate et de Register doit pouvoir être déporté dans un troisième composant (à la racine dans components ?) et appelé au besoin dans CheckBirthdate et dans Register.

J'ai bon ?

Reply:
Le contexte est un poil too much, mais why not, ça fonctionne

Reply:
Je me suis posé la question et j'ai retenu le context parce qu'il n'y avait pas de composant parent aux composants de même niveau. Mes States n'étaient pas mis à jour.

L'alternative, c'était quoi ? 🤔

Reply:
Placer le state dans CheckBirthdate pour les states possibles, sinon dans le parent qui l'appelle puis envoyer en props